src/app/search/query-builder.service.ts

```diff
@@ -191,23 +195,45 @@ export class QueryBuilderService {
    */
   appendQuery(body: any, values: string, oldAdvancedSearchObject: AdvancedSearchObject, searchTerm: boolean): any {
     const advancedSearchObject = { ...oldAdvancedSearchObject };
-    if (values.toString().length > 0) {
-      body = this.searchEverything(body, values);
-    } else {
-      body = body.query('match_all', {});
-    }
-    if (advancedSearchObject) {
+    if (this.hasSettings(advancedSearchObject)) {
       if (advancedSearchObject.searchMode === 'description') {
         this.advancedSearchDescription(body, advancedSearchObject);
       } else if (advancedSearchObject.searchMode === 'files') {
         this.advancedSearchFiles(body, advancedSearchObject);
       }
       values = '';
       searchTerm = false;
+    } else {
+      if (!this.isEmpty(values)) {
+        body = this.searchEverything(body, values);
+      } else {
+        body = body.query('match_all', {});
+      }
     }
     return body;
   }
 
+  private isEmpty(values: string): boolean {
+    return values == undefined || values.toString().trim().length <= 0;
```

Comment:
Nit: `values === undefined`, unless you're trying to catch the null case as well, in which case `!values` is more readable to me.